lib/mark-optional/index.ts

```diff
@@ -1,3 +1,5 @@
+import { OptionalKeys } from "../optional-keys";
+
 export type MarkOptional<Type, Keys extends keyof Type> = Type extends Type
-  ? Omit<Type, Keys> & Partial<Pick<Type, Keys>>
+  ? Partial<Type> & Required<Pick<Type, Exclude<keyof Type, Keys | OptionalKeys<Type & object>>>>
```

Comment:
Combination of `Pick` and `Exclude` is `Omit`

```suggestion
  ? Partial<Type> & Required<Omit<Type, Keys | OptionalKeys<Type>>>
```

WDYT?

Reply:
Yeah makes sense, replaced `Pick` and `Exclude` with `Omit` both in `MarkOptional` and `MarkReadonly`.

Also, removed the `Type extends object` conditional in `MarkReadonly` and replaced it with an intersection (`& object`).

Please refer [this commit](https://github.com/ts-essentials/ts-essentials/pull/418/commits/0919d4cd132e38bb367ff10766684b6222825b65) for changes.

Reply:
> Also, you don't have to use `& object` for `OptionalKeys` now

I kept it because, in future, if we decide to [add a constraint to OptionalKeys](https://github.com/ts-essentials/ts-essentials/pull/416#discussion_r1825716236) to make it consistent with similar types, we wouldn’t have to make any changes here.

But I'll remove it for now, we can worry about it when we add the constraint.

Reply:
Thank you! We will definitely add the generic constraint when it's useful